lib/util/glob.js

```diff
@@ -24,40 +25,39 @@ const IGNORE = typeof Symbol === "function" ? Symbol("ignore") : "_shouldIgnore"
  * @param {function()} shouldIgnore Method to check whether a directory should be ignored.
  * @constructor
  */
-function GlobSync(pattern, options, shouldIgnore) {
-
-    /**
-     * We don't put this thing to argument `options` to avoid
-     * further problems, such as `options` validation.
-     *
-     * Use `Symbol` as much as possible to avoid confliction.
-     */
-    this[IGNORE] = shouldIgnore;
-
-    Sync.call(this, pattern, options);
-}
-
-util.inherits(GlobSync, Sync);
+class GlobSync {
+    constructor(pattern, options, shouldIgnore) {
+
+        /**
+         * We don't put this thing to argument `options` to avoid
+        * further problems, such as `options` validation.
+        *
+        * Use `Symbol` as much as possible to avoid confliction.
+        */
+        this[IGNORE] = shouldIgnore;
+        Sync.call(this, pattern, options);
+    }
 
-/* eslint no-underscore-dangle: ["error", { "allow": ["_readdir", "_mark"] }] */
+    /* eslint no-underscore-dangle: ["error", { "allow": ["_readdir", "_mark"] }] */
+    _readdir(abs, inGlobStar) {
 
-GlobSync.prototype._readdir = function(abs, inGlobStar) {
+        /**
+         * `options.nodir` makes `options.mark` as `true`.
+        * Mark `abs` first
+        * to make sure `"node_modules"` will be ignored immediately with ignore pattern `"node_modules/"`.
 
-    /**
-     * `options.nodir` makes `options.mark` as `true`.
-     * Mark `abs` first
-     * to make sure `"node_modules"` will be ignored immediately with ignore pattern `"node_modules/"`.
+        * There is a built-in cache about marked `File.Stat` in `glob`, so that we could not worry about the extra invocation of `this._mark()`
+        */
+        const marked = this._mark(abs);
 
-     * There is a built-in cache about marked `File.Stat` in `glob`, so that we could not worry about the extra invocation of `this._mark()`
-     */
-    const marked = this._mark(abs);
+        if (this[IGNORE](marked)) {
+            return null;
+        }
 
-    if (this[IGNORE](marked)) {
-        return null;
+        return Sync.prototype._readdir.call(this, abs, inGlobStar);
```

Comment:
This line can be replaced with

```js
return super._readdir(abs, inGlobStar);
```

if `extends Sync` is used in the class declaration.